applications/fishing-map/src/features/map/MapDraw.tsx

```diff
@@ -1,19 +1,291 @@
+    if (features?.length) {
+      setFeatures(features.filter((f, index) => index !== selectedFeatureIndex))
+    }
+  }, [features, selectedFeatureIndex])
+
+  const resetState = useCallback(() => {
+    setLayerName('')
+    setFeatures(undefined)
+    setSelectedFeatureIndex(null)
+    setSelectedEditHandleIndex(null)
+    setNewPointLatitude(undefined)
+    setNewPointLongitude(undefined)
+  }, [])
+
+  const closeDraw = useCallback(() => {
+    resetState()
+    dispatchSetDrawMode('disabled')
+  }, [dispatchSetDrawMode, resetState])
+
+  const createDataset = useCallback(
+    async (features: DrawFeature[], name) => {
+      if (features && features.length > 0) {
+        setLoading(true)
+        const { payload, error } = await dispatchCreateDataset({
+          dataset: getDrawDatasetDefinition(name),
+          file: getFileWithFeatures(name, features),
+          createAsPublic: true,
+        })
+        if (error) {
+          console.warn(error)
+        } else if (payload) {
+          addDataviewFromDatasetToWorkspace(payload)
+        }
+        setLoading(false)
+        closeDraw()
+      }
+    },
+    [addDataviewFromDatasetToWorkspace, closeDraw, dispatchCreateDataset]
+  )
+
+  const onSaveClick = useCallback(() => {
+    if (features && features.length > 0 && layerName) {
+      createDataset(features, layerName)
+    }
+  }, [createDataset, features, layerName])
+
+  const editorMode = useMemo(() => {
+    if (drawMode === 'disabled') {
+      return
+    }
+    return drawMode === 'edit' ? new EditingMode() : new DrawPolygonMode()
+  }, [drawMode])
+
+  const hasFeaturesDrawn = features !== undefined && features.length > 0
+  let saveTooltip = ''
+
+  if (!layerName) {
+    saveTooltip = t('layer.nameRequired', 'Layer name is required')
+  } else if (!hasFeaturesDrawn) {
+    saveTooltip = t('layer.geometryRequired', 'Draw a polygon is required')
+  }
+
   return (
-    <Editor
-      clickRadius={12}
-      mode={mode === 'edit' ? EditingMode : DrawPolygonMode}
-      onSelect={(select: any) => {
-        console.log(select)
-      }}
-    />
+    <Fragment>
+      {editorMode && (
+        <Editor
+          ref={editorRef}
+          clickRadius={12}
+          features={features}
+          mode={editorMode}
+          onUpdate={onEditorUpdate}
+          onSelect={onEditorSelect}
+        />
+      )}
+      {editorMode && currentEditHandle && (
+        <Popup
+          latitude={currentEditHandle[1]}
+          longitude={currentEditHandle[0]}
+          closeButton={true}
+          closeOnClick={true}
+          onClose={() => setSelectedEditHandleIndex(null)}
+          className={cx(styles.popup)}
+          captureClick
+        >
+          <div className={styles.popupContent}>
+            <div className={styles.flex}>
+              <InputText
+                step="0.01"
+                type="number"
+                inputSize="small"
+                value={newPointLatitude || currentEditHandle[1]}
```

Comment:
It's problematic IMHO that the user is not able to enter coordinates with the keyboard - and here's the culprit